docs/blog/2024-11-06-chicory-1.0.0-M1.md

```diff
@@ -0,0 +1,322 @@
+- the first being the index of the beginning of the string in the linear memory
+- the second being the length of such a string
+
+For instance:
+
+```java
+    @WasmExport
+    public int pathFilestatSetTimes(
+            int fd,
+            int lookupFlags,
+            @Buffer String rawPath,
+            long accessTime,
+            long modifiedTime,
+            int fstFlags) {
+        ...
+    }
+```
+
+is effectively translated to the host function `wasi_snapshot_preview1.path_filestat_set_times` with signature:
+- `long fd`
+- `long lookup_flags`
+- **`long raw_path_ptr, long raw_path_size`**
+- `long access_time`
+- `long modified_time`
+- `long fst_flags`
+- returns `long[]`
+
+You can additionally indicate `com.dylibso.chicory.runtime.Instance` and 
+`com.dylibso.chicory.runtime.Memory` as input arguments; for instance:
+
+```java
+@WasmExport
+public int fdFdstatGet(Memory memory, int fd, int buf) {
+    ...
+}
+```
+
+
+## Experimental Modules
+
+We are now clearly marking some modules as `experimental`: while we might still need to introduce a few breaking changes
+here and there before the final 1.0 release *throughout the codebase*, we explicitly call a module
+`experimental` if it is still under development and might be reworked in the near future.
+
+You might notice that some old modules might have changed their name:
+for instance, `aot` is now called `aot-experimental` and the package name has been renamed
+from `com.dylibso.chicory.aot` to `com.dylibso.chicory.experimental.aot`
+
+Nonetheless, we invite you to try them and welcome any feedback.
+
+## Ahead of Time Translation
+
+Chicory has been initially developed as a Java interpreter for WebAssembly; but it has soon acquired
+a bytecode translator from Wasm bytecode to Java bytecode. This is what we call the **Ahead-of-Time** backend.
+The Ahead-of-Time backend is a **drop-in replacement** for the interpreter, and it passes **100% of the same
+spec tests** that the interpreter already supports.
+
+You can instantiate a module by explicitly providing a `MachineFactory`. The default `Machine` implementation is the `InterpreterMachine`. You can opt in to the AoT mode by writing:
+
+```java
+var module = Parser.parse("path/to/your.wasm");
+var instance = Instance.builder(m)
+        .withMatchineFactory(AotMachine::new)
+        .build();
+```
+
+Provide that you add the dependency:
+
+```xml
+      <dependency>
+        <groupId>com.dylibso.chicory</groupId>
+        <artifactId>aot-experimental</artifactId>
+      </dependency>
+```
+
+This will translate every module you instantiate into Java bytecode on-the-fly and in-memory. 
+The resulting code is usually expected to evaluate faster and consuming less memory. This was
+already present in our last release.
+
+With 1.0.0-M1 we are also introducing a new mode of execution: build-time code generation.
+The idea is that this mode of execution will allow you to run Wasm binaries without depending
```

Comment:
right, I meant to say that _eventually_ that would be come true, but it's not necessarily true (nor is it a given that it will _actually_ become true. I will rephrase.

Reply:
> With 1.0.0-M1 we are also introducing a new mode of execution: build-time code generation.
This mode of execution reduces startup time and will remove the need for distributing
the original Wasm binary.